packages/python/plotly/plotly/express/_chart_types.py

```diff
@@ -866,7 +870,13 @@ def choropleth(
     return make_figure(
         args=locals(),
         constructor=go.Choropleth,
-        trace_patch=dict(locationmode=locationmode),
+        trace_patch=dict(
+            locationmode=locationmode,
+            featureidkey=featureidkey,
+            geojson=geojson
+            if not hasattr(geojson, "__geo_interface__")
```

Comment:
add a comment here to explain why this is needed?

Reply:
sure, this is a special hook for `geopandas`... if you have a geopandas frame, it has a special property `.geometry` which has a special `.__geo_interface__` property which returns GeoJSON, so doing this here allows the user to say `px.choropleth(df, geojson=df.geometry,...)`

Reply:
thanks, I was wondering whether this explanation could be added to the source code as a comment, for future maintainers